temporalio/worker/_workflow.py

```diff
@@ -155,6 +160,13 @@ async def run(self) -> None:
         if self._throw_after_activation:
             raise self._throw_after_activation
 
+    def notify_shutdown(self) -> None:
+        if self._could_not_evict_count:
+            logger.warn(
```

Comment:
I wonder if this warrants 'error' level. Sounds like something the user should most definitely know about.

I know in our case the task will just get `SIGKILL'ED` but this is just a red flag in general

Reply:
True, but this could also be while handling an exception which causes shutdown. But I think I agree this maybe should throw.